libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,344 @@
+            variant: evalResp.variant,
+          };
+        }
+      });
+      const listUpdatedFlags = this._getListUpdatedFlags(this._cache, newCache);
+      this._cache = newCache;
+      this._etag = response.httpResponse?.headers.get('etag');
+      return { status: BulkEvaluationStatus.SUCCESS_WITH_CHANGES, flags: listUpdatedFlags };
+    } catch (error) {
+      if (error instanceof OFREPApiTooManyRequestsError) {
+        this._setRetryAfter(error);
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * _getListUpdatedFlags is a function that will compare the old cache with the new cache and
+   * return the list of flags that have been updated / deleted / created.
+   * @param oldCache
+   * @param newCache
+   * @private
+   */
+  private _getListUpdatedFlags(
+    oldCache: { [key: string]: ResolutionDetails<FlagValue> | ResolutionError },
+    newCache: { [key: string]: ResolutionDetails<FlagValue> | ResolutionError },
+  ): string[] {
+    const changedKeys: string[] = [];
+    const oldKeys = Object.keys(oldCache);
+    const newKeys = Object.keys(newCache);
+
+    // Check for added or modified keys in newCache
+    for (const key in newCache) {
+      if (oldKeys.indexOf(key) === -1 || JSON.stringify(oldCache[key]) !== JSON.stringify(newCache[key])) {
+        changedKeys.push(key);
+      }
+    }
+
+    // Check for removed keys in oldCache
+    for (const key in oldCache) {
+      if (newKeys.indexOf(key) === -1) {
+        changedKeys.push(key);
+      }
+    }
+
+    return changedKeys;
+  }
+
+  /**
+   * _setRetryAfter is setting the date until we should stop polling based on the error.
+   * @param error - OFREPApiTooManyRequestsError received
+   * @private
+   */
+  private _setRetryAfter(error: OFREPApiTooManyRequestsError) {
+    const retryAfter = error.response?.headers.get('Retry-After');
+    if (retryAfter) {
+      if (!isNaN(retryAfter as unknown as number)) {
+        this._retryPollingAfter = new Date(Date.now() + (retryAfter as unknown as number) * 1000);
+      } else {
+        const retryAfterDate = new Date(retryAfter);
+        if (retryAfterDate.toString() !== 'Invalid Date') {
+          this._retryPollingAfter = retryAfterDate;
+        }
+      }
+    }
+  }
+  /**
+   * Initialize the provider, it will evaluate the flags and start the polling if it is not disabled.
+   * @param context - the context to use for the evaluation
+   */
+  async initialize?(context?: EvaluationContext | undefined): Promise<void> {
+    try {
+      this._context = context;
+      await this._evaluateFlags(context);
+
+      if (this._pollingInterval > 0) {
+        this.startPolling();
+      }
+
+      this._logger?.debug(`${this.metadata.name} initialized successfully`);
+    } catch (error) {
+      if (error instanceof OFREPApiUnauthorizedError || error instanceof OFREPForbiddenError) {
+        throw new ProviderFatalError('Initialization failed', error);
+      }
+      throw error;
+    }
+  }
```

Comment:
nit: move this closer to the top... for a while I thought it was missing :sweat_smile: